packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -5,55 +5,90 @@
 import { AccessAnalyzer, S3 } from 'aws-sdk'
-import { exec } from 'child_process'
+import { execSync } from 'child_process'
+import { ToolkitError } from '../../shared/errors'
 const AmazonS3URI = require('amazon-s3-uri')
 
+const defaultTerraformConfigPath = path.join(__dirname, '..', 'terraformConfig', 'default.yaml')
+const tempFilePath = path.join(__dirname, 'policyDocument.json')
+const customPolicyCheckDiagnosticCollection = vscode.languages.createDiagnosticCollection('customPolicyCheck')
+const validatePolicyDiagnosticCollection = vscode.languages.createDiagnosticCollection('validatePolicy')
+
 export interface IamPolicyChecksInitialData {
     referenceFilePath: string
-    cfnParameterPath: string
     referenceDocument: string
+    referenceFileErrorMessage: string
+    cfnParameterPath: string
     pythonToolsInstalled: boolean
 }
 
 export class IamPolicyChecksWebview extends VueWebview {
     public static readonly sourcePath: string = 'src/accessanalyzer/vue/index.js'
     public readonly id = 'iamPolicyChecks'
-
-    //private readonly logger = getLogger()
+    private static editedDocumentUri: vscode.Uri
+    private static editedDocumentFileName: string
+    private static editedDocument: string
 
     public constructor(
         private readonly data: IamPolicyChecksInitialData,
         private readonly client: AccessAnalyzer,
+        private readonly region: string,
         public readonly onChangeInputPath = new vscode.EventEmitter<string>(),
         public readonly onChangeReferenceFilePath = new vscode.EventEmitter<string>(),
-        public readonly onChangeCloudformationParameterFilePath = new vscode.EventEmitter<string>()
+        public readonly onChangeCloudformationParameterFilePath = new vscode.EventEmitter<string>(),
+        public readonly onValidatePolicyResponse = new vscode.EventEmitter<[string, string]>(),
+        public readonly onCustomPolicyCheckResponse = new vscode.EventEmitter<[string, string]>(),
+        public readonly onFileReadError = new vscode.EventEmitter<string>()
     ) {
         super(IamPolicyChecksWebview.sourcePath)
         this._setActiveTextEditorListener()
         this._setActiveConfigurationListener()
+        if (vscode.window.activeTextEditor) {
+            IamPolicyChecksWebview.editedDocumentFileName = vscode.window.activeTextEditor.document.uri.path
+            IamPolicyChecksWebview.editedDocument = vscode.window.activeTextEditor.document.getText()
+            IamPolicyChecksWebview.editedDocumentUri = vscode.window.activeTextEditor.document.uri
+        }
     }
 
     public init(): typeof this.data {
         return this.data
     }
 
     public async getReferenceDocument(path: string): Promise<string> {
-        return _getReferenceDocument(path)
+        try {
+            const referenceDocument = await _getReferenceDocument(path)
+            this.onFileReadError.fire('') // Reset error message display if no error is found
```

Comment:
What is the error message that we plan to surface ?

Reply:
The error message is S3 SDK response and `Invalid file path or S3 URI` where file path and S3 URI are invalid.